cupy/cuda/cupy_cudnn.h

```diff
@@ -330,26 +585,33 @@ cudnnStatus_t cudnnRNNBackwardWeights(
          const int seqLength, const cudnnTensorDescriptor_t* xDesc,
          const void* x, const cudnnTensorDescriptor_t hxDesc,
          const void* hx, const cudnnTensorDescriptor_t* yDesc,
-         const void* y, const void* workspace, size_t workSpaceSizeInBytes, 
+         const void* y, const void* workspace, size_t workSpaceSizeInBytes,
          const cudnnFilterDescriptor_t dwDesc, void* dw,
          const void* reserveSpace, size_t reserveSpaceSizeInBytes) {
     return CUDNN_STATUS_NOT_SUPPORTED;
 }
 
-
 #endif // #if CUDNN_VERSION < 5000
 
+
 #if CUDNN_VERSION >= 5000
+// Some functions are renamed in cuDNN v5.
+// Following definitions are for compatibility with cuDNN v5 and higher.
 
 #define cudnnAddTensor_v3 cudnnAddTensor
 #define cudnnConvolutionBackwardData_v3 cudnnConvolutionBackwardData
 #define cudnnConvolutionBackwardFilter_v3 cudnnConvolutionBackwardFilter
 #define cudnnSetConvolutionNdDescriptor_v3 cudnnSetConvolutionNdDescriptor
-
 #define cudnnGetFilterNdDescriptor_v5 cudnnGetFilterNdDescriptor
 
-typedef enum {} cudnnAddMode_t;
+#endif // CUDNN_VERSION >= 5000
+
+
+#if (CUDNN_VERSION >= 5000) || defined(CUPY_NO_CUDA)
```

Comment:
This section code are deleted over cuDNN v5.
This section code are stub for cuDNN v2.